### PR TITLE
feat: check sni and secured for TCP server

### PR DESCRIPTION
**Pull request by benoitgravitee**
## Issue

https://gravitee.atlassian.net/browse/ARCHI-413

## Description

Check SNI and SSL are enabled before creating a TCP server 

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-qyycbxgjfz.chromatic.com)

### CODE REVIEW

File: gravitee-apim-gateway/gravitee-apim-gateway-standalone/gravitee-apim-gateway-standalone-container/src/main/java/io/gravitee/gateway/standalone/vertx/VertxReactorConfiguration.java [hunk 1]
```diff
@@ -91,13 +95,24 @@ public ServerManager serverManager(
...
                 serverManager.register(serverFactory.create(tcpOptions));
             }
         }
 
         return serverManager;
     }
 
+    private void assertTcpOptions(VertxServerOptions options) {
+        if (!options.isSecured() || !options.isSni()) {
+            throw new IllegalArgumentException("Cannot start unsecured TCP server or without SNI enabled");
```
Comment (jhaeyaert) [inline gravitee-apim-gateway/gravitee-apim-gateway-standalone/gravitee-apim-gateway-standalone-container/src/main/java/io/gravitee/gateway/standalone/vertx/VertxReactorConfiguration.java]: Just for clarity, I would rephrase it with

> TCP API requires TLS and SNI to be enabled

**Review by jhaeyaert — APPROVED** [review]
I left a suggestion regarding the clarity of the error message.

**Review by ytvnr — COMMENTED** [review]
it should probably be merged in maintenance branches no ?
